tools/src/main/scala/scala/scalanative/interflow/Eval.scala

```diff
@@ -427,6 +427,12 @@ private[interflow] trait Eval { self: Interflow =>
         eval(value) match {
           case VirtualRef(_, cls, Array(value)) if boxname == cls.name =>
             value
+          case ConvRef(
+                nir.Conv.Bitcast,
+                nir.Rt.Object,
```

Comment:
I think it can be generalized even more, I could have found following NIR that currently would not be handled. The correctness is still guarded by `boxname == cls.name`
```suggestion
                _: nir.Type.RefKind,
```


```LLVM
  1: %4000006 = box[@"T17java.lang.Integer"] %4000003 : int
  1: %4000007 = bitcast[@"T17java.lang.Integer"] %4000006 : !?@"T17java.lang.Integer"
  1: %4000008 = unbox[@"T17java.lang.Integer"] %4000007 : @"T17java.lang.Integer"
```